### ISSUE TITLE
Unaligned float access with FFI on armeabi-v7a (Android) causes SIGBUS

### ISSUE THREAD

**Issue by xefinity** [LuaJIT/LuaJIT#978]
Example Lua code that causes SIGBUS (BUS_ADRALN):
```lua
ffi = require 'ffi'  
ffi.cdef[[ 
#pragma pack(push, 1) 
 
typedef struct Test {
  char c; 
  float f; 
} Test;

#pragma pack(pop)
]]

t = ffi.new("Test")
print("before unaligned access")
t.f = 42
print("after unaligned access")
```

Backtrace:
```
#0  0x0046279c in lj_cconv_ct_ct (cts=cts@entry=0xf7c7abe8, d=d@entry=0xf7c80dc0,s=0xf7c80d80, dp=dp@entry=0xf7c840d1 "", sp=0xf7c7a298 "*", flags=2) at lj_cconv.c:252
#1  0x00462b94 in lj_cconv_ct_tv (cts=0xf7c7abe8, d=0xf7c80dc0, dp=0xf7c840d1 "", o=<optimized out>, flags=2) at lj_cconv.c:643
#2  0x00461f74 in lj_cdata_set (cts=0x2a, cts@entry=0xf7c7abe8, d=0x338, dp=0x4621e4 <lj_cconv_ct_ct+164> "\220\001", o=0x0, o@entry=0xf7c7a298, qual=<optimized out>) at lj_cdata.c:301
#3  0x0046f3cc in lj_cf_ffi_meta___newindex (L=0xf7c791c8) at lib_ffi.c:178
#4  0x0040e754 in lj_BC_FUNCC ()
#5  0x004246bc in lua_pcall (L=0x2a, L@entry=0xf7c791c8, nargs=nargs@entry=0, nresults=<optimized out>, errfunc=errfunc@entry=1) at lj_api.c:1145
#6  0x0040b438 in docall (L=L@entry=0xf7c791c8, narg=0, clear=0) at luajit.c:122
#7  0x0040b220 in dotty (L=L@entry=0xf7c791c8) at luajit.c:254
#8  0x0040ace0 in pmain (L=0xf7c791c8) at luajit.c:134
#9  0x0040e754 in lj_BC_FUNCC ()
#10 0x004246f0 in lua_cpcall (L=0x2a, L@entry=0xf7c791c8, func=0x338, ud=0x4621e4 <lj_cconv_ct_ct+164>) at lj_api.c:1173
#11 0x0040a308 in main (argc=1, argv=<optimized out>) at luajit.c:581
```

LuaJIT version: 505e2c0
Compilation flags: ```-O2 -fomit-frame-pointer -mfloat-abi=softfp```, built with NDK r25c.

**Comment by MikePall** [comment]
Why are you opening an issue for this? The FFI is a low-level interface. If unaligned accesses crash in plain C, so will they with the FFI.

tl;dr: If it hurts, don't do it!
